### PR TITLE
Don't run play tests in build branch accept

### PR DESCRIPTION
**Pull request by donat**
This PR resolves https://github.com/gradle/gradle-private/issues/1011#issuecomment-389476371

The ignored tests are executed as part of the master accept, so we can remove them from quick feedback and from the build branch accept stage.

This is my first take on the solution. I had a couple of tries to modify the TC configuration, but we have too many things tangled together. Maybe we should revisit how we declare our build types, especially with the new DSL introduced with the latest TC release.

### CODE REVIEW

**Comment by marcphilipp** [comment]
This means we are removing the following test coverage for `platformPlay` that was previously part of Quick Feedback/Branch Accept stages, right?

- Java 8 Oracle Linux
- Java 7 Oracle Windows
- Java 7 Oracle Linux
- Java 8 Oracle Windows

Master Accept will run it on Java 10 Oracle Linux and Java 7 IBM Linux.

**Comment by big-guy** [comment]
@donat from Marc's comment above, does this remove the coverage for Java7/8 on Linux/Windows or just move it to a later stage? I think we can delay where the coverage runs, but we shouldn't remove it entirely.

**Comment by blindpirate** [comment]
What if I changed the code of "slow project" and want to verify it? Can I switch it on somewhere?

**Comment by donat** [comment]
@marcphilipp I've added a a modification to defer the marked build types (e.g.`platformPlay`) to master accept. Unfortunately I can't really test it with a current setup, so I can only verify once my change got merged.

I serously don't like the current state. The modeling elements and the declaration of the TC is convoluted (configuration is declared in constructors!), which makes it really hard to introduce anything new.

File: .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt [hunk 1]
```diff
@@ -91,14 +94,53 @@ class CIConfigIntegrationTests {
...
+                        TestCoverage(TestType.parallel, OS.windows, JvmVersion.java7)),
+                    omitsSlowProjects = false)
+            ),
+            subProjects = listOf(
+                GradleSubproject("fastBuild"),
+                GradleSubproject("slowBuild", containsSlowTests = true)
+            )
+        )
+        val p = RootProject(m)
+        printTree(p)
```
Comment (marcphilipp) [inline .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt]: Can we add an assertion here? The goal is for `slowBuild` to be in `Stage3`, right?

File: .teamcity/Gradle_Check/model/CIBuildModel.kt [hunk 2]
```diff
@@ -147,10 +98,59 @@ data class CIBuildModel (
...
+                    performanceTests = listOf(
+                            PerformanceTestType.experiment)),
+            Stage("Historical Performance", "Once a week: Run performance tests for multiple Gradle versions",
+                    trigger = Trigger.weekly,
+                    performanceTests = listOf(
+                            PerformanceTestType.historical)),
+            Stage("Experimental", "On demand: Run experimental tests",
+                    trigger = Trigger.never,
+                    runsIndependent = true,
+                    functionalTests = listOf(TestCoverage(TestType.platform, OS.linux, JvmVersion.java11))))
```
Comment (marcphilipp) [inline .teamcity/Gradle_Check/model/CIBuildModel.kt]: Why was this moved down?

File: .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt (hunk 1 above)
Comment (donat) [reply .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt]: Fixed

File: .teamcity/Gradle_Check/model/CIBuildModel.kt (hunk 2 above)
Comment (donat) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: Actually, it was only moved into the constructor, so that I could overwrite the values in the tests. I've restored the same ordering.
